src/linker/Linker/MSBuildMessageContainer.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Mono.Linker
+{
+	public readonly struct MSBuildMessageContainer
+	{
+		/// <summary>
+		/// Can either be the tool name (illinker, monolinker) or a filename with
+		/// parenthesized information about the line and column that triggered the
+		/// linker to output an error (or warning) message.
+		/// </summary>
+		public MessageOrigin Origin { get; }
+		
+		public MessageCategory Category { get; }
+
+		/// <summary>
+		/// Further categorize the message.
+		/// </summary>
+		public string Subcategory { get; }
+
+		/// <summary>
+		/// Code identifier for errors and warnings reported by the IL linker.
+		/// </summary>
+		public int Code { get; }
+		
+		/// <summary>
+		/// Optional user friendly text describing the error or warning.
+		/// </summary>
+		public string Text { get; }
+
+		public MSBuildMessageContainer (
+			int code,
```

Comment:
Please fix the order, text is the most important and should never be empty and not last in argument